### PR TITLE
Validate new password with password policy regex

### PR DESCRIPTION
**Pull request by mpmadhavig**
This effort will get the new password policy from the password policy connector and validates the new passwords.

### Related Issues
- https://github.com/wso2/product-is/issues/15386

### Related PRs
- None

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/users/components/user-change-password.tsx [hunk 1]
```diff
@@ -487,86 +521,87 @@ export const ChangePasswordComponent: FunctionComponent<ChangePasswordPropsInter
...
-                   <Grid.Row>
-                       <Grid.Column mobile={ 16 } tablet={ 16 } computer={ 14 }>
-                           <Message
-                               type="warning"
-                               content={ t("console:manage.features.user.modals.changePasswordModal.message") }
-                           />
-                       </Grid.Column>
-                   </Grid.Row>
-               </>
+                                        }
```
Comment (hwupathum) [inline apps/console/src/features/users/components/user-change-password.tsx:591]: remove the additional spaces

File: apps/console/src/features/users/components/user-change-password.tsx [hunk 2]
```diff
@@ -199,7 +215,7 @@ export const ChangePasswordComponent: FunctionComponent<ChangePasswordPropsInter
                     }
                 }
             ],
-            "schemas": ["urn:ietf:params:scim:api:messages:2.0:PatchOp"]
+            "schemas": [ "urn:ietf:params:scim:api:messages:2.0:PatchOp" ]
```
Comment (hwupathum) [inline apps/console/src/features/users/components/user-change-password.tsx:218]: remove the additional spaces

File: apps/console/src/features/users/components/user-change-password.tsx [hunk 3]
```diff
@@ -313,47 +329,49 @@ export const ChangePasswordComponent: FunctionComponent<ChangePasswordPropsInter
...
-                   </Grid.Row>
-               </>
-           );
+            return (
+                <>
+                    <Grid.Row>
+                        <Grid.Column mobile={ 16 } tablet={ 16 } computer={ 14 }>
+                            <p>
+                                Following are the password reset options available. Please make sure
+                                you have enabled the required configurations.
```
Comment (hwupathum) [inline apps/console/src/features/users/components/user-change-password.tsx:354]: remove the additional spaces

File: apps/console/src/features/users/components/user-change-password.tsx [hunk 4]
```diff
@@ -487,86 +521,87 @@ export const ChangePasswordComponent: FunctionComponent<ChangePasswordPropsInter
...
-                               label={ t(
-                                   "console:manage.features.user.forms.addUserForm.inputs.newPassword.label"
-                               ) }
-                               name="newPassword"
-                               placeholder={ t(
-                                   "console:manage.features.user.forms.addUserForm.inputs." +
+                <>
+                    <Grid.Row>
+                        <Grid.Column mobile={ 16 } tablet={ 16 } computer={ 14 }>
+                            <Field
```
Comment (hwupathum) [inline apps/console/src/features/users/components/user-change-password.tsx:527]: remove the additional spaces

File: apps/console/src/features/users/components/user-change-password.tsx (hunk 2 above)
Comment (mpmadhavig) [reply apps/console/src/features/users/components/user-change-password.tsx:218]: These are required by the ES-lint code style formatter

File: apps/console/src/features/users/components/user-change-password.tsx [hunk 5]
```diff
@@ -136,7 +152,7 @@ export const ChangePasswordComponent: FunctionComponent<ChangePasswordPropsInter
         setPasswordResetOption("setPassword");
     };
 
-    const passwordResetOptions = [
+    const passwordResetOptions: any = [
```
Comment (thivi) [inline apps/console/src/features/users/components/user-change-password.tsx:155]: We should be able to use `RadioChild` here.

File: apps/console/src/features/users/components/user-change-password.tsx [hunk 6]
```diff
@@ -219,7 +235,7 @@ export const ChangePasswordComponent: FunctionComponent<ChangePasswordPropsInter
             handleCloseChangePasswordModal();
             handleUserUpdate(user.id);
         })
-            .catch((error) => {
+            .catch((error: any) => {
```
Comment (thivi) [inline apps/console/src/features/users/components/user-change-password.tsx:238]: Let's specify the error type here.

File: apps/console/src/features/users/components/user-change-password.tsx [hunk 7]
```diff
@@ -188,7 +204,7 @@ export const ChangePasswordComponent: FunctionComponent<ChangePasswordPropsInter
             return;
         }
 
-        const data = {
+        const data: any = {
```
Comment (thivi) [inline apps/console/src/features/users/components/user-change-password.tsx:207]: I think there should be an interface for this object.

**Review by savindi7 — APPROVED** [review]

**Pull request merged**